sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -5,183 +5,116 @@
-
-// @public
-export interface ContextAPI {
-    active(): Context;
-}
+export function createTracingClient(options: TracingClientOptions): TracingClient;
 
 // @public
-export function createSpanFunction(args: CreateSpanFunctionArgs): <T extends {
-    tracingOptions?: OperationTracingOptions | undefined;
-}>(operationName: string, operationOptions?: T | undefined, startSpanOptions?: SpanOptions | undefined) => {
-    span: Span;
-    updatedOptions: T;
-};
+export function createTracingContext(options?: CreateTracingContextOptions): TracingContext;
 
 // @public
-export interface CreateSpanFunctionArgs {
-    namespace: string;
-    packagePrefix: string;
+export interface CreateTracingContextOptions {
+    client?: TracingClient;
+    namespace?: string;
+    parentContext?: TracingContext;
+    span?: TracingSpan;
 }
 
 // @public
-export type Exception = ExceptionWithCode | ExceptionWithMessage | ExceptionWithName | string;
-
-// @public
-export interface ExceptionWithCode {
-    code: string | number;
-    message?: string;
-    name?: string;
-    stack?: string;
+export interface Instrumenter {
+    createRequestHeaders(spanId: TracingSpanIdentifier): Record<string, string>;
+    parseTraceparentHeader(traceparentHeader: string): TracingSpanIdentifier | undefined;
+    startSpan(name: string, spanOptions?: InstrumenterSpanOptions): {
+        span: TracingSpan;
+        tracingContext: TracingContext;
+    };
+    withContext<CallbackArgs extends unknown[], Callback extends (...args: CallbackArgs) => ReturnType<Callback>>(context: TracingContext, callback: Callback, callbackThis?: ThisParameterType<Callback>, ...callbackArgs: CallbackArgs): ReturnType<Callback>;
 }
 
 // @public
-export interface ExceptionWithMessage {
-    code?: string | number;
-    message: string;
-    name?: string;
-    stack?: string;
+export interface InstrumenterSpanOptions extends TracingSpanOptions {
+    packageInformation: {
+        name: string;
+        version?: string;
+    };
+    tracingContext?: TracingContext;
 }
 
 // @public
-export interface ExceptionWithName {
-    code?: string | number;
-    message?: string;
-    name: string;
-    stack?: string;
+export interface OperationTracingOptions {
+    tracingContext?: TracingContext;
 }
 
 // @public
-export function extractSpanContextFromTraceParentHeader(traceParentHeader: string): SpanContext | undefined;
-
-// @public
-export function getSpan(context: Context): Span | undefined;
-
-// @public
-export function getSpanContext(context: Context): SpanContext | undefined;
-
-// @public
-export function getTraceParentHeader(spanContext: SpanContext): string | undefined;
-
-// @public
-export function getTracer(): Tracer;
-
-// @public
-export function getTracer(name: string, version?: string): Tracer;
-
-// @public
-export type HrTime = [number, number];
-
-// @public
-export function isSpanContextValid(context: SpanContext): boolean;
+export type SpanStatus = {
+    status: "success";
```

Comment:
otel default status is unset and we should have it too and use it (e.g. all http 4xx are unset)

Reply:
Because you cannot read a span's status, only set it, I was thinking this can represent the statuses that can be explicitly set in an operation. As an implementation detail, the status will initially be "unset" in OTel. Is there a scenario where a user will want to explictly set status as "unset" or "none"?